### PR TITLE
Topchiy - Lesson 7 Done

### PR DESCRIPTION
**Pull request by sergiy2303**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -26,3 +26,10 @@ $(document).on('change', 'input#product_image', function(){
   };
   reader.readAsDataURL(input.files[0]);
 })
+
+$(document).on('change keyup', 'input.search-field', function() {
+  var value = $(this).val();
+  $.get("/products/search", {search: value}, function(data) {
```
Comment (Alexey-Lukin) [inline app/assets/javascripts/application.js]: application paths should not be hardcoded in javascript

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+class ProductsController < ApplicationController
+
+  def index
+    @session = Session.new
+    if params[:id]
+      @products = products.where(category_id: params[:id])
```
Comment (Alexey-Lukin) [inline app/controllers/products_controller.rb]: You should fetch correct collection of objects based on request params and then paginate this collection

Comment (Alexey-Lukin) [inline app/controllers/products_controller.rb]:
```
  @products = if params[:id]
                Category.find(params[:id]).products
              else
                           
              end
  then paginate 
```

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    end
+    flash.now[:notice] = 'No products for this category' if @products.empty?
+  end
+
+  def search
+    if params[:search].present?
+      @products = Product.page(params[:page]).per(9).search(params[:search]).records
+    else
+      @products = products
+    end
```
Comment (Alexey-Lukin) [inline app/controllers/products_controller.rb]: search, then paginate

File: app/controllers/products_controller.rb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      @products = products.where(category_id: params[:id])
+    else
+      @products = products
+    end
+    flash.now[:notice] = 'No products for this category' if @products.empty?
+  end
+
+  def search
+    if params[:search].present?
+      @products = Product.page(params[:page]).per(9).search(params[:search]).records
```
Comment (Alexey-Lukin) [inline app/controllers/products_controller.rb]: why default per_page value - `10` does not work for you?

File: app/controllers/products_controller.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    @products = Product.page(params[:page]).per(9)
+  end
+
+  def product
+    @product ||= Product.new
+  end
+  helper_method :product
+
+  def current_product
+    @product = Product.find_by(id: params[:id])
```
Comment (Alexey-Lukin) [inline app/controllers/products_controller.rb]: `Product.find(params[:id])`

File: app/views/products/_products.html.slim [hunk 6]
```diff
@@ -0,0 +1,13 @@
+.row
+  - @products.each do |product|
+    .col-sm-6.col-md-4
+      .thumbnail
+        =image_tag product.image
```
Comment (Alexey-Lukin) [inline app/views/products/_products.html.slim]: add space after `=`

File: app/views/products/index.html.slim [hunk 7]
```diff
@@ -0,0 +1,16 @@
+div class='content'
```
Comment (Alexey-Lukin) [inline app/views/products/index.html.slim]: `.content`

File: lib/tasks/elasticsearch.rake [hunk 8]
```diff
@@ -0,0 +1 @@
+require 'elasticsearch/rails/tasks/import'
```
Comment (Alexey-Lukin) [inline lib/tasks/elasticsearch.rake]: remove weird sign at the end of line

File: app/controllers/products_controller.rb (hunk 4 above)
Comment (sergiy2303) [inline app/controllers/products_controller.rb]: @Alexey-Lukin  becouse in bootstrap for Thumbnail uses col-md-4 as default (3 columns).

File: app/controllers/products_controller.rb (hunk 5 above)
Comment (sergiy2303) [inline app/controllers/products_controller.rb]: :+1:

File: app/views/products/_products.html.slim (hunk 6 above)
Comment (sergiy2303) [inline app/views/products/_products.html.slim]: @Alexey-Lukin 
:+1:

File: lib/tasks/elasticsearch.rake (hunk 8 above)
Comment (sergiy2303) [inline lib/tasks/elasticsearch.rake]: @Alexey-Lukin I am not sure that we really need this file. I created it by readme on github.

**Pull request closed without merging**